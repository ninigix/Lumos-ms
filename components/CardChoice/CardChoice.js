import React from "react";
import { TouchableOpacity, View } from "react-native";

import MyText from "../MyText/MyText";
import IconButton from "../IconButton/IconButton";

import styles from "./CardChoice.style";

const CardChoice = ({
  leftComponent,
  rightComponent,
  handleOnPress,
  leftComponentAdditionalLine,
  iconType,
  iconIsActive,
  iconBackgroundColor
}) => {
  const renderText = () => (
    <MyText textStyle={{ fontSize: 16 }}>{leftComponentAdditionalLine}</MyText>
  );

  const renderBoldText = () => (
    <MyText isBold textStyle={{ fontSize: 16 }}>
      {leftComponent}
    </MyText>
  );

  const renderLeftContent = () => (
    <TouchableOpacity
      onPress={handleOnPress}
      style={styles.leftContentWrapper}
      disabled={!handleOnPress}
    >
      <IconButton
        type={iconType}
        wasSelected={iconIsActive}
        backgroundColor={iconBackgroundColor}
      />
      {leftComponentAdditionalLine ? (
        <View
          style={{
            display: "flex",
            flexDirection: "column"
          }}
        >
          {renderText()}
          {renderBoldText()}
        </View>
      ) : (
        renderBoldText()
      )}
    </TouchableOpacity>
  );

  return (
    <View style={styles.wrapper}>
      <View style={styles.leftWrapper}>{renderLeftContent()}</View>
      <View style={styles.rightWrapper}>
        <View style={styles.greyBackground}>{rightComponent}</View>
      </View>
    </View>
  );
};

export default CardChoice;
