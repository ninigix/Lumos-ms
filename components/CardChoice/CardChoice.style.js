import {StyleSheet} from "react-native";

export default StyleSheet.create({
    leftWrapper: {
        flex: 5,
        justifyContent: "center",
        alignItems: "center",
        borderTopLeftRadius: 4,
        borderBottomLeftRadius: 4
    },
    rightWrapper: {
        flex: 1,
        display: "flex",
        minWidth: "15%"
    },
    wrapper: {
        display: "flex",
        height: 70,
        borderRadius: 4,
        borderWidth: 0.5,
        borderColor: "#d6d7da",
        flexDirection: "row",
        margin: 15
    },
    greyBackground: {
        flex: 1,
        backgroundColor: "#EBEBE4",
        borderTopRightRadius: 4,
        borderBottomRightRadius: 4,
        justifyContent: "center",
        alignItems: "center"
    },
    leftContentWrapper: {
        display: "flex",
        flexDirection: "row",
        width: "100%",
        alignItems: "center"
    }
});
