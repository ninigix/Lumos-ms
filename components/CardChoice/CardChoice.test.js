import renderer from "react-test-renderer";
import React from "react";

import CardChoice from "./CardChoice";

it("renders correctly", () => {
  const tree = renderer.create(<CardChoice />).toJSON();
  expect(tree).toMatchSnapshot();
});
