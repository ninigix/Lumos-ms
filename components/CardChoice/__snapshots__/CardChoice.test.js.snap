// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "#d6d7da",
      "borderRadius": 4,
      "borderWidth": 0.5,
      "display": "flex",
      "flexDirection": "row",
      "height": 70,
      "margin": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomLeftRadius": 4,
        "borderTopLeftRadius": 4,
        "flex": 5,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EBEBE4",
            "borderRadius": 4,
            "display": "flex",
            "height": 40,
            "justifyContent": "center",
            "margin": 20,
            "width": 40,
          }
        }
      >
        <Image
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "fontFamily": "raleway-bold",
            "fontSize": 16,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "minWidth": "15%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EBEBE4",
          "borderBottomRightRadius": 4,
          "borderTopRightRadius": 4,
          "flex": 1,
          "justifyContent": "center",
        }
      }
    />
  </View>
</View>
`;
